components/formats-bsd/src/loci/formats/tiff/TiffSaver.java

```diff
@@ -1021,13 +1021,22 @@ private void makeValidIFD(IFD ifd, int pixelType, int nChannels) {
     if (nChannels == 1 && ifd.getIFDValue(IFD.COLOR_MAP) != null) {
       pi = PhotoInterp.RGB_PALETTE;
     }
-    else if (nChannels == 3) {
+    else if (nChannels == 3 || nChannels == 4) {
+      // TODO: handle CMYK; nChannels == 4, extraSamples 0 (can be omitted)
       if (ifd.getIFDValue(IFD.COMPRESSION).equals(TiffCompression.JPEG.getCode())) {
         // see https://github.com/ome/bioformats/issues/3856
         pi = PhotoInterp.Y_CB_CR;
+        if (nChannels == 4) {
+          int[] extraSamplesArray = {2};
+          ifd.putIFDValue(IFD.EXTRA_SAMPLES, extraSamplesArray);
+        }
       }
       else {
         pi = PhotoInterp.RGB;
+        if (nChannels == 4) {
+          int[] extraSamplesArray = {1};
```

Comment:
I don't think it's safe to assume here that the 4th channel is always what the TIFF specification refers to as "associated alpha data", which is what the `1` here indicates. That may be true in many cases, but I'm uncomfortable with this assumption for *every* 4-channel RGB input. As noted in the previous review, I think setting the `ExtraSamples` entry to 0 ("unspecified data" per the TIFF specification) is the safest option.